frontend/public/components/service-classes.tsx

```diff
@@ -0,0 +1,118 @@
+import { SectionHeading, detailsPage, navFactory, ResourceLink, ResourceSummary } from './utils';
+// eslint-disable-next-line no-unused-vars
+import { K8sResourceKind, K8sResourceKindReference } from '../module/k8s';
+import * as _ from 'lodash-es';
+
+const ServiceClassesReference: K8sResourceKindReference = 'ClusterServiceClass';
+
+const normalizeIconClass = (iconClass) => {
+  return _.startsWith(iconClass, 'icon-') ? `font-icon ${iconClass}` : iconClass;
+};
+
+const ServiceClassHeader: React.SFC<ServiceClassHeaderProps> = props => <ListHeader>
+  <ColHead {...props} className="col-lg-4 col-md-4 col-sm-6 col-xs-6" sortField="metadata.name">Name</ColHead>
+  <ColHead {...props} className="col-lg-1 col-md-1 col-sm-1 hidden-xs">Icon</ColHead>
+  <ColHead {...props} className="col-lg-4 col-md-4 col-sm-5 col-xs-6" sortField="spec.externalMetadata.displayName">Display Name</ColHead>
+  <ColHead {...props} className="col-lg-3 col-md-3 hidden-sm hidden-xs" sortField="spec.externalName">External Name</ColHead>
+</ListHeader>;
+
+const ServiceClassListRow: React.SFC<ServiceClassesRowProps> = ({obj: serviceClass}) => <ResourceRow obj={serviceClass}>
+  <div className="col-lg-4 col-md-4 col-sm-6 col-xs-6 co-resource-link-wrapper">
+    <ResourceLink kind="ClusterServiceClass" name={serviceClass.metadata.name} />
+  </div>
+  <div className="col-lg-1 col-md-1 col-sm-1 hidden-xs">
+    <span className={`fa-2x ${normalizeIconClass(_.get(serviceClass, ['spec', 'externalMetadata', 'console.openshift.io/iconClass']))}`} />
+  </div>
+  <div className="col-lg-4 col-md-4 col-sm-5 col-xs-6">
+    {serviceClass.spec.externalMetadata.displayName}
+  </div>
+  <div className="col-lg-3 col-md-3 hidden-sm hidden-xs">{serviceClass.spec.externalName}</div>
+</ResourceRow>;
+
+const ServiceClassDetails: React.SFC<ServiceClassDetailsProps> = ({obj: serviceClass}) => <React.Fragment>
+  <div className="co-m-pane__body">
+    <SectionHeading text="Service Class Overview" />
+    <div className="row">
+      <div className="col-sm-6">
+        <ResourceSummary resource={serviceClass} showPodSelector={false} showNodeSelector={false}>
+          <dt>Display Name</dt>
+          <dd>{_.get(serviceClass, 'spec.externalMetadata.displayName', '-')}</dd>
+          <dt>External Name</dt>
+          <dd>{_.get(serviceClass, 'spec.externalName', '-')}</dd>
+          <dt>Provider</dt>
+          <dd>{_.get(serviceClass, 'spec.externalMetadata.providerDisplayName', '-')}</dd>
+          {~_.isEmpty('serviceClass.spec.tags') && <React.Fragment>
+            <dt>Tags</dt>
+            <dd>{_.join(_.get(serviceClass, 'spec.tags', '-'), ', ')}</dd>
+          </React.Fragment>
+          }
+        </ResourceSummary>
+      </div>
+      <div className="col-sm-6 ">
+        <dd className="co-service-class-title">
+          <span className={`fa-5x co-service-class-title__icon ${normalizeIconClass(_.get(serviceClass, ['spec', 'externalMetadata', 'console.openshift.io/iconClass']))}`} />
```

Comment:
Need to handle the case when there's no icon.

It's probably worth having a `ServiceClassIcon` component that we can reuse with all the logic for icon class vs image URL.